[OOB_5_4_RC1] Compile error in BLE_URIBeacon for all toolchain

## How to reproduce

```
$ git clone -b oob_test_mbed-os-5.4 https://github.com/ARMmbed/mbed-os-example-ble
$ cd mbed-os-example-ble\BLE_URIBeacon
$ mbed deploy
$ mbed compile -m NRF52_DK -t ARM
```


```
(snip)
[Error] main.cpp@96,0:  #20: identifier "DFUService" is undefined
[Error] main.cpp@96,0:  #20: identifier "dfu" is undefined
[Error] main.cpp@96,0:  #79: expected a type specifier
[ERROR] "./mbed-os/platform/toolchain.h", line 23: Warning:  #1215-D: #warning directive: toolchain.h has been replaced by mbed_toolchain.h, please update to mbed_toolchain.h [since mbed-os-5.3]
".\source\main.cpp", line 96: Error:  #20: identifier "DFUService" is undefined
".\source\main.cpp", line 96: Error:  #20: identifier "dfu" is undefined
".\source\main.cpp", line 96: Error:  #79: expected a type specifier
.\source\main.cpp: 1 warning, 3 errors

[mbed] ERROR: "python" returned error code 1.
[mbed] ERROR: Command "python -u C:\Users\toywat01\Documents\mbed\oob\mbed-os-example-ble\BLE_URIBeacon\mbed-os\tools\make.py -t ARM -m nrf52_dk --source . --build .\BUILD\nrf52_dk\ARM" in "C:\Users\toywat01\Documents\mbed\oob\mbed-os-example-ble\BLE_URIBeacon"

```

mbed ls

```
BLE_URIBeacon (None)
|- mbed-os (3a27568a505b)
`- shields\TARGET_ST_BLUENRG (a5d5d0b55839)
```

I got same compile error by IAR and GCC_ARM toolchain.

--- a/BLE_URIBeacon/source/main.cpp
+++ b/BLE_URIBeacon/source/main.cpp
@@ -18,7 +18,6 @@
 #include "mbed.h"
 #include "ble/BLE.h"
 #include "ble/services/URIBeaconConfigService.h"
-#include "ble/services/DFUService.h"
 #include "ble/services/DeviceInformationService.h"
 #include "ConfigParamsPersistence.h"
 
@@ -92,8 +91,6 @@ int main()
         error("failed to accommodate URI");
     }
 
-    // Setup auxiliary services to allow over-the-air firmware updates, etc
-    DFUService *dfu = new DFUService(ble);
     DeviceInformationService *deviceInfo = new DeviceInformationService(ble, "ARM", "UriBeacon", "SN1", "hw-rev1", "fw-rev1", "soft-rev1");
 
     ble.startAdvertising(); /* Set the whole thing in motion. After this call a GAP central can scan the URIBeaconConfig

--- a/BLE_URIBeacon/source/nrfConfigParamsPersistence.cpp
+++ b/BLE_URIBeacon/source/nrfConfigParamsPersistence.cpp
@@ -14,11 +14,6 @@
  * limitations under the License.
  */
 
-extern "C" {
-#include "pstorage.h"
-}
-
-#include "nrf_error.h"
 #include "ConfigParamsPersistence.h"
 
 /**
@@ -42,6 +37,14 @@ struct PersistentParams_t {
  * pass it on to the 'pstorage' APIs.
  */
 static PersistentParams_t persistentParams;
+
+#if defined(TARGET_NORDIC)
+
+extern "C" {
+#include "pstorage.h"
+}
+
+#include "nrf_error.h"
 
 static pstorage_handle_t pstorageHandle;
 
@@ -104,3 +107,28 @@ void saveURIBeaconConfigParams(const URIBeaconConfigService::Params_t *paramsP)
                         0 /* offset */);
     }
 }
+
+#else 
+
+/* In memory implementation */
+bool loadURIBeaconConfigParams(URIBeaconConfigService::Params_t *paramsP)
+{
+    static bool pstorageInitied = false;
+    if (!pstorageInitied) {
+        memset(&persistentParams.params, 0, sizeof(URIBeaconConfigService::Params_t));
+        memset(paramsP, 0, sizeof(URIBeaconConfigService::Params_t));
+        pstorageInitied = true;
+        return false;
+    }
+
+    memcpy(paramsP, &persistentParams.params, sizeof(URIBeaconConfigService::Params_t));
+    return true;
+}
+
+/* In memory implementation */
+void saveURIBeaconConfigParams(const URIBeaconConfigService::Params_t *paramsP)
+{
+	memcpy(&persistentParams.params, paramsP, sizeof(URIBeaconConfigService::Params_t));
+}
+
+#endif 

